src/storage/createIndexedDBStorage.ts

```diff
@@ -0,0 +1,550 @@
+							initialQueries = initialQueries.filter((query) => !isFilter(query));
+						}
+
+						let cursorRequest: IDBRequest;
+						const filterChain = initialFilters[0] && initialFilters[0].filterChain;
+						let { firstFilter, secondFilter } = getFilters(filterChain);
+						if (filterChain && !filterChain.length) {
+							initialFilters.shift();
+						}
+						let { keyRange, index, firstFilterUsed, secondFilterUsed } = getKeyRangeAndIndex(
+							state, objectStore, firstFilter, secondFilter
+						);
+						if (!secondFilterUsed && secondFilter) {
+							initialFilters.unshift(secondFilter);
+						}
+						if (!firstFilterUsed && firstFilter) {
+							initialFilters.unshift(firstFilter);
+						}
+						if (keyRange && index) {
+							cursorRequest = index.openCursor(keyRange);
+						}
+						else {
+							cursorRequest = objectStore.openCursor();
+						}
+
+						cursorRequest.onsuccess = (event: any) => {
+							const cursor = event.target.result;
+							if (cursor) {
+								// Apply filters and if the new item is not filtered, apply other
+								// queries to the collection after adding it in.
+								const newItems = initialFilters.reduce((prev, next) => {
+									if (prev.length) {
+										return next.apply(prev);
+									} else {
+										return prev;
+									}
+								}, [ cursor.value ]);
+								if (newItems.length) {
+									items = initialQueries.reduce(
+										(prev, next) => next.apply(prev), [ ...items, newItems[0] ]
+									);
+								}
+								cursor.continue();
+							}
+							else {
+								resolve(inMemoryQuery ? inMemoryQuery.apply(items) : items);
+							}
+						};
+					});
+				}
+			);
+			fetchResult.totalLength = totalLength;
+			return fetchResult;
+
+		},
+
+		get(this: Storage<{}, {}>, ids: string[]): Promise<{}[]> {
+			const state = instanceStateMap.get(this);
+			return state.dbInitializationPromise.then(() => {
+				const { objectStore } = getTransactionAndObjectStore(state);
+				return Promise.all(ids.map((id) => {
+					const request = objectStore.get(id);
+					return createRequestPromise(request).then(function() {
+						return request.result;
+					});
+				})).then((data) => data.filter((item) => Boolean(item)));
+			});
+		},
+
+		put(this: Storage<{}, {}>, items: {}[], options?: CrudOptions): Promise<UpdateResults<{}>> {
+			const state = instanceStateMap.get(this);
+			return state.dbInitializationPromise.then(() => {
+				const { transactionPromise, objectStore } = getTransactionAndObjectStore(state, true);
+				const rejectOverwrite = options && options.rejectOverwrite;
+				const ids = this.identify(items);
+				return Promise.all(items.map((item, index) => {
+					const id = ids[index];
+					return createRequestPromise(
+						rejectOverwrite ? objectStore.add(item, id) : objectStore.put(item, id), transactionPromise
+					).then(() => item);
+				})).then((updatedItems) => ({
```

Comment:
Fixed. I know this style is more consistent with having `else` on a new line, and understand the reasoning for it, but I just wanted to note that I find it very awkward to read.